controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -445,6 +445,25 @@ func inspectionDisabled(host *metal3v1alpha1.BareMetalHost) bool {
 	return false
 }
 
+// clearInspectAnnotation deletes inspect.metal3.io annotation exist on the provided host
+func clearInspectAnnotation(host *metal3v1alpha1.BareMetalHost) (dirty bool) {
+	delete(host.Annotations, inspectAnnotationPrefix)
+	dirty = true
+	return dirty
+}
+
+// hasInspectAnnotation checks for existence of inspect.metal3.io annotation
+// and returns true if it exist
+func hasInspectAnnotation(host *metal3v1alpha1.BareMetalHost) bool {
+	annotations := host.GetAnnotations()
+	if annotations != nil {
+		if _, ok := annotations[inspectAnnotationPrefix]; ok {
```

Comment:
Judging by line 442, the annotation can be `disabled`, in which case we don't want to trigger inspection?

Reply:
Good catch! Updating it Thanks.